ios/browser/api/url/url_formatter.h

```diff
@@ -0,0 +1,83 @@
+
+NS_SWIFT_NAME(URLFormatter.SchemeDisplay)
+typedef NS_ENUM(NSUInteger, BraveURLSchemeDisplay) {
+  BraveURLSchemeDisplay_Show,
+  BraveURLSchemeDisplay_OmitHttpAndHttps,
+  /// Omit cryptographic (i.e. https and wss).
+  BraveURLSchemeDisplay_OnitCryptographic,
+};
+
+OBJC_EXPORT
+NS_SWIFT_NAME(URLFormatter)
+@interface BraveURLFormatter: NSObject
++ (bool)isPunycode:(NSString*)domain;
++ (NSString*)formatUrlForSecurityDisplay:(NSURL*)origin schemeDisplay:(BraveURLSchemeDisplay)schemeDisplay;
++ (NSString*)formatUrlForDisplayOmitSchemePathAndTrivialSubdomains:(NSURL*)url;
+@end
+
+
+// MARK: - Spoof Checker
+
+/// The |SkeletonType| and |TopDomainEntry| are mirrored in trie_entry.h. These
+/// are used to insert and read nodes from the Trie.
+/// The type of skeleton in the trie node.
+NS_SWIFT_NAME(URLSpoofChecker.SkeletonType)
+typedef NS_ENUM(NSUInteger, BraveSpoofCheckerSkeletonType) {
+  BraveSpoofCheckerSkeletonType_Full,
+  BraveSpoofCheckerSkeletonType_SeparatorsRemoved
+};
+
+NS_SWIFT_NAME(URLSpoofChecker.LookalikeURLMatchType)
+typedef NS_ENUM(NSUInteger, BraveSpoofCheckerLookalikeURLMatchType) {
+  BraveSpoofCheckerLookalikeURLMatchType_None = 0,
+  /// DEPRECATED: Use kSkeletonMatchTop500 or kSkeletonMatchTop5k.
+  /// kTopSite = 1,
+  BraveSpoofCheckerLookalikeURLMatchType_SiteEngagement = 2,
+  BraveSpoofCheckerLookalikeURLMatchType_EditDistance = 3,
+  BraveSpoofCheckerLookalikeURLMatchType_EditDistanceSiteEngagement = 4,
+  BraveSpoofCheckerLookalikeURLMatchType_TargetEmbedding = 5,
+  BraveSpoofCheckerLookalikeURLMatchType_SkeletonMatchTop500 = 6,
+  BraveSpoofCheckerLookalikeURLMatchType_SkeletonMatchTop5k = 7,
+  BraveSpoofCheckerLookalikeURLMatchType_TargetEmbeddingForSafetyTips = 8,
+
+  /// The domain name failed IDN spoof checks but didn't match a safe hostname.
+  /// As a result, there is no URL to suggest to the user in the form of "Did
+  /// you mean <url>?".
+  BraveSpoofCheckerLookalikeURLMatchType_FailedSpoofChecks = 9,
+
+  BraveSpoofCheckerLookalikeURLMatchType_CharacterSwapSiteEngagement = 10,
+  BraveSpoofCheckerLookalikeURLMatchType_CharacterSwapTop500 = 11
+};
+
+NS_SWIFT_NAME(URLSpoofChecker.TopDomainEntry)
+@interface URLSpoofCheckerTopDomainEntry: NSObject
+/// The domain name.
+@property (nonatomic, readonly) NSString* domain;
+/// True if the domain is in the top 500.
+@property (nonatomic, readonly) bool isTop500;
+/// Type of the skeleton stored in the trie node.
+@property (nonatomic, readonly) BraveSpoofCheckerSkeletonType skeletonType;
+@end
+
+NS_SWIFT_NAME(URLSpoofChecker)
+@interface BraveURLSpoofChecker: NSObject
++ (URLSpoofCheckerTopDomainEntry*)getSimilarTopDomain:(NSString*)hostname;
++ (URLSpoofCheckerTopDomainEntry*)lookupSkeletonInTopDomains:(NSString*)hostname;
++ (NSArray<NSString*>*)getSkeletons:(NSURL*)url;
++ (bool)isLookalikeURL:(NSURL*)url urlMatchType:(BraveSpoofCheckerLookalikeURLMatchType*)urlMatchType suggestedURL:(NSURL* _Nonnull* _Nullable)suggestedURL;
```

Comment:
Make a new type that contains results and return that (or nil if it fails) instead of messing with setting pointers (for example https://developer.apple.com/documentation/foundation/nstextcheckingresult)